public/app/features/variables/adhoc/reducer.test.ts

```diff
@@ -0,0 +1,205 @@
+import { reducerTester } from '../../../../test/core/redux/reducerTester';
+import cloneDeep from 'lodash/cloneDeep';
+import { getVariableTestContext } from '../state/helpers';
+import { toVariablePayload } from '../state/types';
+import { adHocVariableReducer, filterAdded, filterRemoved, filterUpdated, filtersRestored } from './reducer';
+import { VariablesState } from '../state/variablesReducer';
+import { AdHocVariableModel, AdHocVariableFilter } from '../../templating/variable';
+import { createAdHocVariableAdapter } from './adapter';
+
+describe('adHocVariableReducer', () => {
+  const adapter = createAdHocVariableAdapter();
+
+  describe('when filterAdded is dispatched', () => {
+    it('then state should be correct', () => {
+      const uuid = '0';
+      const { initialState } = getVariableTestContext(adapter, { uuid });
+      const filter = createFilter('a');
+      const payload = toVariablePayload({ uuid, type: 'adhoc' }, filter);
+
+      reducerTester<VariablesState>()
+        .givenReducer(adHocVariableReducer, cloneDeep(initialState))
+        .whenActionIsDispatched(filterAdded(payload))
+        .thenStateShouldEqual({
+          [uuid]: {
+            ...initialState[uuid],
+            filters: [filter],
```

Comment:
nit: what are your thoughts about always keeping the thenStateShouldEqual as static as possible? I don't mind the `...initialState[uuid]` part, but the things we assert on should be actual values and not reusing variables. What do you think?
```suggestion
            filters: [{key:'adsasdads', value:'asdasdasd'}],
```

Reply:
hmm, I am guessing that you are afraid that tests (now or later when changing them) will get false positives?

Reply:
Yes, that's it

Reply:
OK ok, will have a look at this one as well! Great feedback! <3